### PR TITLE
New commander x16 emulator

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/commander-x16/emulator.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, lib
+, fetchFromGitHub
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/commander-x16/emulator.nix:2]:
```suggestion
{ stdenv
, lib
, fetchFromGitHub
```

File: pkgs/misc/emulators/commander-x16/emulator.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  buildInputs = [ SDL2 ];
+
+  installPhase = ''
+    runHook preInstall
+    install -D --mode 755 --target-directory $out/bin/ x16emu
+    install -D --mode 444 --target-directory $out/share/doc/${pname} README.md
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/commander-x16/emulator.nix:28]: meta should be last.

File: pkgs/misc/emulators/commander-x16/rom.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, lib
+, fetchFromGitHub
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/commander-x16/rom.nix:2]:
```suggestion
{ stdenv
, lib
, fetchFromGitHub
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111000` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>cc65: log <a href="https://termbin.com/xthun">https://termbin.com/xthun</a></li>
<li>x16-rom: log was empty</li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>x16-emulator</li>
</ul>
</details>

File: pkgs/misc/emulators/commander-x16/emulator.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/misc/emulators/commander-x16/emulator.nix:28]: Usually, meta is the last attribute.
However, in the rare cases when `passthru` is used, it comes last.

(The idea of passthru here is in the case someone creates a wrapping script encapsulating the emulator and the rom; according to the upstream site, they are updated at the same time, and older ROMs may not work with newer emulators.)

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111000` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>x16-emulator</li>
<li>x16-rom</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111000` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>cc65: log <a href="https://termbin.com/uzdx">https://termbin.com/uzdx</a></li>
<li>x16-rom: log was empty</li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>x16-emulator</li>
</ul>
</details>

**Pull request merged**